numpy/core/tests/test_multiarray.py

```diff
@@ -6487,6 +6487,22 @@ def test_roundtrip_scalar(self):
         # Issue #4015.
         self._check_roundtrip(0)
 
+    def test_invalid_buffer_format(self):
+        # datetime64 cannot be used fully in a buffer yet
+        # Should be fixed in the next Numpy major release
+        dt = np.dtype([('a', int), ('b', 'M8[s]')])
+        dt_tag = np.dtype({'names':['a','b'], 'formats':['<i8','u1'],
+                           'offsets':[0,8], 'itemsize':16})
+        a = np.empty(3, dt)
+        m = memoryview(a)
```

Comment:
I'd like to see an explicit `as_dict` test here to be sure that the buffer is actually valid